### PR TITLE
add silent failure log

### PR DESCRIPTION
**Pull request by cloudmagic80**
## Summary

- *This work is behind a feature toggle (flipper): NO
Log ZSF for IVC forms

https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/practices/zero-silent-failures/logging-silent-failures.md

Use this log message when your team has identified a silent failure AND has made that failure known to the user via an email notification via VA Notify.

Screenshot 2024-11-13 at 7 27 54 AM

### CODE REVIEW

File: modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb [hunk 1]
```diff
@@ -10,19 +10,27 @@ def update_status
         Datadog::Tracing.trace('Start PEGA Status Update') do
           data = JSON.parse(params.to_json)
 
+          tags = ['service:ivc_champva', 'function:form submission to Pega']
```
Comment (michaelclement) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb:13]: I believe this should be
`tags = ['service:veteran-ivc-champva-forms', 'function:form submission to Pega']`

to match our service tag in `ivc_champva/v1/application_controller.rb`

Comment (cloudmagic80) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb:13]: Ahh good catch

**Review by stevelong00 — APPROVED** [review]
Looks great!

File: modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb (hunk 1 above)
Comment (rmtolmach) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb]: @cloudmagic80 do you need to add the form id in the messaging like @michaelclement suggests?

Comment (cloudmagic80) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb]: Yes I've made the changes

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**